### DISCUSSION TITLE
rocketmq5 proxy 的兼容性问题

### DISCUSSION THREAD

**Discussion by pxd98** [apache/rocketmq#7919] [Q&A] [answered]
我的服务端是5.2.0，客户端是4.9.1，在发送消息的时候报了 15 Request doesn't have field bname的报错，我看proxy里的源码是需要判断请求里是否存在bname这个字段
```
if (request.getCode() == RequestCode.SEND_MESSAGE_V2) {
            if (request.getExtFields().get(BROKER_NAME_FIELD_FOR_SEND_MESSAGE_V2) == null) {
                return RemotingCommand.buildErrorResponse(ResponseCode.VERSION_NOT_SUPPORTED,
                    "Request doesn't have field bname");
            }
            brokerName = request.getExtFields().get(BROKER_NAME_FIELD_FOR_SEND_MESSAGE_V2);
        } else {
            if (request.getExtFields().get(BROKER_NAME_FIELD) == null) {
                return RemotingCommand.buildErrorResponse(ResponseCode.VERSION_NOT_SUPPORTED,
                    "Request doesn't have field bname");
            }
            brokerName = request.getExtFields().get(BROKER_NAME_FIELD);
        }
```
我排查下来是因为4.9.1版本的SendMessageRequestHeader中缺少了requestHeader.setBname(mq.getBrokerName());这段代码和RpcRequestHeader这个父类。
我修改了这部分代码并在createSendMessageRequestHeaderV2中增加了字段n后就可以正常发送消息了，但是我看兼容性文档中说rocketmq5是完全兼容rocketmq4.9版本的，所以想问一下在不升级客户端的情况下是否有其他方式发送消息或者说只能升级客户端版本才行，谢谢。
具体的修复是不是在这里完成的？[[ISSUE #3905] Support bname in protocol for 4.9.x client](https://github.com/apache/rocketmq/pull/5161)

**Comment by leizhiyuan** [comment, marked as answer]
升级到4.9.x 最新版本，低版本不兼容

**Reply by pxd98** [reply]
好的，谢谢

**Reply by PHK-20** [reply]
用的是github.com/apache/rocketmq-client-go/v2 如何解决这个问题呢

**Reply by fanlv** [reply]
@PHK-20 想问下最后是怎么解决的？

**Reply by fanlv** [reply]
知道了，master 已经修复了，但是没打 tag。要 go get github.com/apache/rocketmq-client-go/v2@master  就好了。
